### PR TITLE
Simplify logic for picking root category

### PR DESCRIPTION
**Pull request by rmilecki**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Previous logic grew into a bit complex one. It had 2 conditional parts each checking for selected `BLOCK_CATEG_ROOT_CATEGORY` values. They were checking for different setups and later one was conditionally modifying result of the first one.<br><br>All of that made `getWidgetVariables()` code a bit hard to follow (understand) and hard to extend (add more scenarios).<br><br>This change cleans that up. It adds a `switch()` and handles each `BLOCK_CATEG_ROOT_CATEGORY` value with a separated code. It should make code easier to understand, debug & develop.
| Type?         | refacto
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |
| How to test?  | Open BO and navigate to the `ps_categorytree` module configuration. Select every "Category root" available option one by one and verify in FO they work as expected. This PR does not modify the behavior, it modifies how the code is written.

### CODE REVIEW

**Comment by rmilecki** [comment]
@Hlavtox: I really like your work on dropping cookie usage in the https://github.com/PrestaShop/ps_categorytree/pull/67 .

As for `getWidgetVariables()` logic however I think that handling each `BLOCK_CATEG_ROOT_CATEGORY` value separately would make code cleaner.

Would you be kind to look at my proposed change, please? Would you agree my logic with `switch()` solution makes this code any cleaner?

**Review by 0x346e3730 — APPROVED** [review]

**Review by matks — APPROVED** [review]
Thank you @rmilecki nice PR 👍 

I'm asking QA team if they can verify we changed no behavior

**Review by MhiriFaten — APPROVED** [review]
Hello @rmilecki ,

I have checked every "Category root" available option one by one and they work as expectedin FO. :tada:  
So it's QA approved :heavy_check_mark: 

Thank you

**Comment by nicosomb** [comment]
Thank you @rmilecki !

**Comment by rmilecki** [comment]
Thank you!
